src/consensus/aft/raft.h

```diff
@@ -803,7 +823,7 @@ namespace aft
       else
       {
         if (
-          can_endorse_primary() && ticking &&
+          !is_retired_committed() && ticking &&
```

Comment:
A node can run for election until all future primaries are aware its retirement is committed. This feels a little too helpful, I think we could restrict to just seeing the retirement committed, because at least one node in the new config should be eligible to become primary at this point, and move commit to that point.

I want to see if this fixes the scenarios in #5961 first, before making this condition more restrictive if necessary.